adapters/mqtt-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/mqtt/quarkus/Application.java

```diff
@@ -83,16 +96,73 @@ void onStop(final @Observes ShutdownEvent ev) {
 
     private VertxBasedMqttProtocolAdapter adapter() {
 
+        final DeviceRegistrationClient registrationClient = registrationClient();
+        final DeviceConnectionClient deviceConnectionClient = deviceConnectionClient();
+        final CommandTargetMapper commandTargetMapper = commandTargetMapper();
+
+        commandTargetMapper.initialize(new CommandTargetMapperContext() {
+
+            @Override
+            public Future<List<String>> getViaGateways(
```

Comment:
It seems to me, the content of this method can be reduced to just:
````
        final VertxBasedMqttProtocolAdapter adapter = new VertxBasedMqttProtocolAdapter();
        setCollaborators(adapter);
        adapter.setConfig(adapterProperties);
        adapter.setMetrics(metrics);
        adapter.setMessageMapping(messageMapping());
        return adapter;
````
using the parent `setCollaborators(AbstractProtocolAdapterBase)` method.

Same goes for the corresponding class/method in the `hono-adapter-http-vertx-quarkus` module.

Reply:
That's true and that is what I wanted to do. Don't know why it ended up this way ...